File: src/main/scala/exoskeleton/common/ExoConfiguration.scala
package exoskeleton.common

import java.io.File

import net.minecraftforge.common.config.Configuration

object ExoConfiguration{
  var adv_vision: Boolean = true;
  var xray_color: Int = 0xFF3399;
  private var config: Configuration = null;

  def init(file: File): Unit ={
    this.config = new Configuration(file);
    this.config.load();
    this.adv_vision = this.config.getBoolean("adv_vision", "settings", true, "Enable for texture overrides");
    this.config.save();
  }

  def save(): Unit ={
    this.config.save();
  }
}

File: src/main/scala/exoskeleton/common/EXOConfiguration.java
package exoskeleton.common;

import net.minecraftforge.common.config.Configuration;

import java.io.File;

public final class EXOConfiguration{
    public static int xray_color = 0xFF3399;

    private static EXOConfiguration instance;
    private Configuration config;

    public static EXOConfiguration instance(){
        return instance == null ? instance = new EXOConfiguration() : instance;
    }

    public void initialize(File file){
        this.config = new Configuration(file);
        this.config.load();
        EXOConfiguration.xray_color = Integer.parseInt(this.config.getString("xray_color", "settings", "#FFFFFF", "Change the XRay line colour").substring(1), 16);
        this.config.save();
    }
}